packages/block-editor/src/hooks/style.js

```diff
@@ -77,11 +77,16 @@ export function getInlineStyles( styles = {} ) {
 		const subPaths = STYLE_PROPERTY[ propKey ].properties;
 		// Ignore styles on elements because they are handled on the server.
 		if ( has( styles, path ) && 'elements' !== first( path ) ) {
-			if ( !! subPaths ) {
-				subPaths.forEach( ( suffix ) => {
-					output[
-						propKey + capitalize( suffix )
-					] = compileStyleValue( get( styles, [ ...path, suffix ] ) );
+			// Checking if style value is a string allows for shorthand css
+			// option and backwards compatibility for border radius support.
+			const styleValue = get( styles, path );
+
+			if ( !! subPaths && ! isString( styleValue ) ) {
+				Object.entries( subPaths ).forEach( ( entry ) => {
+					const [ name, suffix ] = entry;
```

Comment:
Can it be the case that the block attributes only contain a subset of the longhand values (e.g.: margin-top, margin-bottom)? If so, we'd need to cover against that (not creating an inline style if a longhand such as margin-right doesn't exist).

Reply:
The block attributes can contain only a subset of the longhand values as you described. 

The code prior to this PR, created inline styles for all the defined subpaths as well. I believe they are just omitted when the inline styles are applied. Thankfully, that means we can tighten up this code without causing block validation errors. 

I've updated this in https://github.com/WordPress/gutenberg/pull/31641/commits/706321be7b993e8223f141e7b9573d11d745af4f

Reply:
I've tested this and the UI controls don't allow to set only some of the longhands: they'll need to be null or will be reset to the previous value. I've tested this by tweaking the block markup manually -to simulate a situation in which the UI control allows you to do that- and it works fine.